ui/lib/core/addon/components/info-table-item-array.js

```diff
@@ -15,63 +14,81 @@ import { isWildcardString } from 'vault/helpers/is-wildcard-string';
+ * @param [itemRoute=vault.cluster.secrets.backend.show] - {string} - Tells what route the link should go to when selecting the individual item.
+ * @param [modelType] {string} - Tells which model you want data for the allOptions to be returned from.  Used in conjunction with the the isLink.
  * @param [wildcardLabel] {String} - when you want the component to return a count on the model for options returned when using a wildcard you must provide a label of the count e.g. role.  Should be singular.
  * @param [queryParam] {String} - If you want to specific a tab for the View All XX to display to.  Ex: role
  * @param [backend] {String} - To specify which backend to point the link to.
  * @param [viewAll] {String} - Specify the word at the end of the link View all xx.
  */
-export default Component.extend({
-  layout,
-  'data-test-info-table-item-array': true,
-  attributeBindings: ['data-test-info-table-item-array'],
-  allOptions: null,
-  displayArray: null,
-  wildcardInDisplayArray: false,
-  store: service(),
-  displayArrayAmended: computed('displayArray', function () {
+export default class InfoTableItemArray extends Component {
+  @tracked allOptions = null;
+  @tracked wildcardInDisplayArray = false;
+  @service store;
+
+  get rootRoute() {
+    return this.args.rootRoute || 'vault.cluster.secrets.backend.list-root';
+  }
+
+  get itemRoute() {
+    return this.args.itemRoute || 'vault.cluster.secrets.backend.show';
+  }
+
+  get displayArray() {
+    return this.args.displayArray || null;
+  }
+
+  get displayArrayAmended() {
     let { displayArray } = this;
-    if (!displayArray) {
-      return;
-    }
+    if (!displayArray) return null;
     if (displayArray.length >= 10) {
       // if array greater than 10 in length only display the first 5
       displayArray = displayArray.slice(0, 5);
     }
-
     return displayArray;
-  }),
+  }
 
-  checkWildcardInArray: task(function* () {
+  @task *checkWildcardInArray() {
     if (!this.displayArray) {
       return;
     }
     let filteredArray = yield this.displayArray.filter((item) => isWildcardString(item));
-    this.set('wildcardInDisplayArray', filteredArray.length > 0 ? true : false);
-  }).on('didInsertElement'),
 
-  fetchOptions: task(function* () {
-    if (this.isLink && this.modelType) {
+    this.wildcardInDisplayArray = filteredArray.length > 0 ? true : false;
+  }
+
+  @task *fetchOptions() {
+    if (this.args.isLink && this.args.modelType) {
       let queryOptions = {};
 
-      if (this.backend) {
-        queryOptions = { backend: this.backend };
+      if (this.args.backend) {
+        queryOptions = { backend: this.args.backend };
       }
 
-      let options = yield this.store.query(this.modelType, queryOptions);
+      let options = yield this.store.query(this.args.modelType, queryOptions);
       this.formatOptions(options);
     }
-  }).on('didInsertElement'),
+    this.checkWildcardInArray.perform();
```

Comment:
chaining the preforms because they need to be called on didInsertElement which is no longer a hook in components. We are using the ember-render-modifiers third-party addon which allows us to use did-insert inside the hbs file. I call the first concurrency task, and at the end of that, call the second. The perform hbs helper won't allow me to pass two concurrency tasks, hence chaining.

Reply:
Do we need concurrency here? Wondering if these can be changed to actions since these are one action happening after another. (This doesn't get re-called when the options change, right?)